parser/xml/src/test/java/org/codice/ddf/parser/xml/TestXmlParserConfigurator.java

```diff
@@ -0,0 +1,117 @@
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+import java.util.List;
+
+import javax.xml.bind.ValidationEvent;
+import javax.xml.bind.ValidationEventHandler;
+import javax.xml.bind.annotation.adapters.XmlAdapter;
+
+import org.codice.ddf.parser.ParserConfigurator;
+import org.junit.Before;
+import org.junit.Test;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+public class TestXmlParserConfigurator {
+    private ParserConfigurator pc;
+
+    private ValidationEventHandler testHandler;
+
+    private XmlAdapter testAdapter;
+
+    @Before
+    public void setup() {
+        pc = new XmlParserConfigurator();
+
+        testHandler = new ValidationEventHandler() {
+            @Override
+            public boolean handleEvent(ValidationEvent event) {
+                return false;
+            }
+        };
+
+        testAdapter = new XmlAdapter() {
+            @Override
+            public Object unmarshal(Object v) throws Exception {
+                return null;
+            }
+
+            @Override
+            public Object marshal(Object v) throws Exception {
+                return null;
+            }
+        };
+    }
+
+    @Test
+    public void testContextPath() {
+        List<String> inputPaths = ImmutableList.of("hello", "world", "goodnight", "columbus");
+
+        pc.setContextPath(inputPaths);
+        List<String> contextPath = pc.getContextPath();
+
+        assertEquals(inputPaths.size(), contextPath.size());
+
+        for (int i = 0; i < inputPaths.size(); i++) {
+            assertEquals(inputPaths.get(i), contextPath.get(i));
+        }
+    }
+
+    @Test
+    public void testClassLoader() {
+        pc.setClassLoader(null);
+        assertNull(pc.getClassLoader());
+
+        pc.setClassLoader(TestXmlParserConfigurator.class.getClassLoader());
+        assertEquals(TestXmlParserConfigurator.class.getClassLoader(), pc.getClassLoader());
+    }
+
+    @Test
+    public void testValidationEventHandler() {
+        pc.setHandler(testHandler);
+        assertEquals(testHandler, pc.getHandler());
+    }
+
+    @Test
+    public void testAdapter() {
+        pc.setAdapter(testAdapter);
+        assertEquals(testAdapter, pc.getAdapter());
+    }
+
+    @Test
+    public void testProperties() {
+        assertNotNull(pc.getProperties());
+        assertEquals(0, pc.getProperties().size());
+
+        pc.addProperty("aaa", 123);
+        assertEquals(1, pc.getProperties().size());
+        assertEquals(123, pc.getProperties().get("aaa"));
+        pc.addProperty(null, "This will not add");
+        assertEquals(1, pc.getProperties().size());
+
+        pc.addProperties(ImmutableMap.<String, Object>of("bbb", 2, "ccc", 3));
+        assertEquals(3, pc.getProperties().size());
+        assertTrue(pc.getProperties().containsKey("aaa"));
```

Comment:
I would suggest checking out AssertJ.  It has nice support for testing collections.

https://joel-costigliola.github.io/assertj/